e2e/test/MessageReceiveE2ETests.cs

```diff
@@ -28,63 +32,147 @@ public MessageReceiveE2ETests()
+        {
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Http1, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceReceiveSingleMessageAbandon_Http()
+        {
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Http1, VerifyReceivedC2DMessageAndAbandon).ConfigureAwait(false);
         }
 
         [TestMethod]
-        public async Task Message_DeviceReceiveSingleMessage_Http()
+        public async Task Message_DeviceReceiveSingleMessageReject_Http()
         {
-            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Http1).ConfigureAwait(false);
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Http1, VerifyReceivedC2DMessageAndReject).ConfigureAwait(false);
         }
 
         [TestMethod]
         public async Task X509_DeviceReceiveSingleMessage_Amqp()
         {
-            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
+            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Amqp_Tcp_Only, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
         }
 
         [TestMethod]
         public async Task X509_DeviceReceiveSingleMessage_AmqpWs()
         {
-            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Amqp_WebSocket_Only).ConfigureAwait(false);
+            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Amqp_WebSocket_Only, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
         }
 
         [TestMethod]
         public async Task X509_DeviceReceiveSingleMessage_Mqtt()
         {
-            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Mqtt_Tcp_Only).ConfigureAwait(false);
+            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Mqtt_Tcp_Only, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
         }
 
         [TestMethod]
         public async Task X509_DeviceReceiveSingleMessage_MqttWs()
         {
-            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Mqtt_WebSocket_Only).ConfigureAwait(false);
+            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Mqtt_WebSocket_Only, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
         }
 
         [TestMethod]
         public async Task X509_DeviceReceiveSingleMessage_Http()
         {
-            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Http1).ConfigureAwait(false);
+            await ReceiveSingleMessage(TestDeviceType.X509, Client.TransportType.Http1, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceReceiveMultipleMessages_Amqp()
```

Comment:
nit: naming - same as above

Reply:
This was left due to some testing, test name is right, added right call.